### PR TITLE
Change video compression parameters

### PR DESCRIPTION
**Pull request by Gargron**
The goal is to get rid of unused/meaningless parameters and increase the quality.

### CODE REVIEW

File: lib/paperclip/transcoder.rb [hunk 1]
```diff
@@ -38,8 +41,11 @@ def make
         @output_options['vframes'] = 1
       when 'mp4'
         unless eligible_to_passthrough?(metadata)
-          @output_options['acodec'] = 'aac'
-          @output_options['strict'] = 'experimental'
+          bitrate = (metadata.width * metadata.height * 30 * BITS_PER_PIXEL) / 1_000
+
+          @output_options['b:v']     = "#{bitrate}k"
+          @output_options['maxrate'] = "#{bitrate + 192}k"
+          @output_options['bufsize'] = "#{bitrate * 2}k"
```
[the remark this answers is not in the document]
Comment (Gargron) [reply lib/paperclip/transcoder.rb:48]: Would it have an effect on ffmpeg's memory usage? That might be a limited resource on some setups.

File: app/models/media_attachment.rb [hunk 2]
```diff
@@ -98,16 +98,10 @@ class MediaAttachment < ApplicationRecord
...
-        'vsync' => 'cfr',
+        'preset' => 'veryfast',
         'c:v' => 'h264',
-        'maxrate' => '1300K',
-        'bufsize' => '1300K',
-        'b:v' => '1300K',
-        'frames:v' => 60 * 60 * 3,
-        'crf' => 18,
+        'c:a' => 'aac',
+        'b:a' => '192k',
```
[the remark this answers is not in the document]
Comment (Gargron) [reply app/models/media_attachment.rb:104]: Are the passthrough options merged with the original options? I'm not seeing where that would happen. To me it looks like only the passthrough options are used during passthrough...

[the remark this answers is not in the document]
Comment (ClearlyClaire) [reply app/models/media_attachment.rb:104]: Sorry, you're right on this.

File: app/models/media_attachment.rb [hunk 3]
```diff
@@ -162,7 +156,7 @@ class MediaAttachment < ApplicationRecord
 
   VIDEO_CONVERTED_STYLES = {
     small: VIDEO_STYLES[:small].freeze,
-    original: VIDEO_FORMAT.freeze,
+    original: VIDEO_FORMAT.deep_merge(convert_options: { 'frames:v' => 10_800 }).freeze, # Limit the total number of frames that can be snuck through a GIF
```
[the remark this answers is not in the document]
Comment (Gargron) [reply app/models/media_attachment.rb:159]: So 10,800 frames is about 3 minutes at 60fps and 6 minutes at 30fps. I feel like it's maybe a bit unintuitive to have a duration limit that is frame-based. So if we could get rid of it, I'd be up for it.

[the remark this answers is not in the document]
Comment (ClearlyClaire) [reply app/models/media_attachment.rb:159]: I'm not sure it's needed anymore now that we ditched `vsync: cfr`, but I'd still be careful with that, transcoding is complicated and I'm sure there are many edge cases I fail to envision.

Also, a 3 minutes 60fps video already sounds pretty long for a microblogging platform, but I guess we can up the limit. My server is aware of very few videos longer than that, most of those are jokes (e.g. the entire 1995 Hackers movie encoded to an unrecognizable pixel mess).

What about a limit of `36_000` (5 minutes at 120fps)?

**Review by ClearlyClaire — APPROVED** [review]

**Comment by smiba** [comment]
Just a quick sanity check, but wouldn't this potentially result in massive file sizes? 

https://github.com/mastodon/mastodon/blob/5d93e98da433b76ba39bbe9f1f36814742e47230/lib/paperclip/transcoder.rb#L44

If we max out the limits defined in the following file, at 10 minutes of 60FPS:
https://github.com/mastodon/mastodon/blob/5d93e98da433b76ba39bbe9f1f36814742e47230/app/models/media_attachment.rb#L45-L47

You'd get a file that's 1.71GB in size? 

bitrate = (3840 * 2160 * 30 * 0.11) / 1000 = 27371 kbps

27371 kbps * (30000 / 60) = 13685500Kb = 1710687KB = ~1.71GB

I might be entirely wrong, I also haven't tested it, but I always believed the previous fixed bitrate was simply used to ensure a file wouldn't get above the max file size.

**Comment by Gargron** [comment]
@smiba You're right. Changing it in #26970

**Pull request merged**